### ISSUE TITLE
Open Stage Control freezes

### ISSUE THREAD

**Issue by cyberic99** [jean-emmanuel/open-stage-control#131]
On git ed2f35d649161

O-S-C freezes sometimes (often with my workflow)

I'm sorry but I do not have lots of debug info nor a repro case...
I can reproduce it often, but it requires specific hardware and software...
(we'll do a demo soon)

If I have a chrome debugger opened before the freeze, I see an uninterrupted flow of this message:
```
[Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive.
[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive.

jquery.slim.min.js:3
```
And the widgets become completely unresponsive.
I'd like to provide more info, is there a way to prevent js minification or to get source map files?

EDIT: I still see the OSC received in the console, in debug mode. But no error message in the console.
Is there a way to display a stack trace?

Thank you
Eric

**Comment by jean-emmanuel** [comment]
If there's no error in the console, there can't be any stack trace: the violation message you quote is a harmless warning. Source map files are already generated, but they only help when an actual error is logged. Although, jQuery source map is not bundled, because it's not likely to help here (bugs are more likely to be found in the code I write !). Actually, I'm currently working towards removing jquery, at least from perf-sensitive operations (see https://github.com/jean-emmanuel/open-stage-control/tree/dev).

Freezes often come from infinite loop errors, but these usually get logged after some time, I'm gonna need more informations to be able to help:

- client browser (embedded / remote device type & browser version)
- o-s-c run command / config
- session file

**Comment by cyberic99** [comment]
In fact the errors in the console never stop, and I get roughly 
Chromium 56.0.2924.76 Built on 8.7, running on Debian 8.7
In fact these 'violation' messages never stop, or sometimes they are displayed at a very high rate (>1000/s)

I'll try to reproduce with a simple case...
It cole be /EDIT commands

**Comment by cyberic99** [comment]
Wow, it was hard to reproduce...

So here is a way to reproduce it every time on my PC:
in o-s-c sources, git ed2f35d:
$ npm start 

open attached file

press F12

launch several times (5 times at least):
```
send_osc osc.udp://localhost:8080  /EDIT/MERGE 'services' '{"variables": {"midiplayer": "192.168.0.47:5001"}}'
send_osc osc.udp://localhost:8080  /EDIT 'services' '{"variables": {"ip": "192.168.0.47:5000", "osc": "192.168.0.47:8080"}}'
```

It seems something is exponential here...
After 5 times, I get lots of errors

![screenshot from 2018-01-16 01 01 05](https://user-images.githubusercontent.com/193917/34966010-6a2a6ca2-fa50-11e7-98ff-6620acbc0e48.png)

and each time I relaunch these commands, error sum up and o-s-c freezes and eats 100% CPU...


I tried to change some of the variable names, like midiplayer -> xxx, and... I couldn't reproduce the issue! it only works with these exact values...

Thank you for your help
Eric

**Comment by jean-emmanuel** [comment]
Ok I reproduced the issue (with the `2tabs.json` from your previous issue): while these are not errors, the exponential behaviour shows some kind of memory leak : old widgets elements where getting new events listeners instead of being simply removed. It seems to be fixed in the `dev` branch. As for these warning messages, they're almost gone ;)

**Comment by cyberic99** [comment]
thx for taking time to reproduce (and fix) this issue!
I hope we'll be able to show all the strange stuff we do with o-s-c ;-)
I'll try the dev branch tonight.

**Comment by jean-emmanuel** [comment]
Looking forward to seeing that :)
If you have a multitouch device, I could use some feedback on the dev branch btw.

**Comment by cyberic99** [comment]
not yet.
For now I use the mouse only.
But if you like I can test o-s-c on an android tablet as well.

**Comment by jean-emmanuel** [comment]
Actually the warnings can't be avoided, the behaviour described in the messages is normal.

**Comment by cyberic99** [comment]
ok but why do I get so much occurence of this messages, and why is it increasing every time I relaunch my app?

**Comment by jean-emmanuel** [comment]
What do you mean by relaunching the app ?  Can you describe precisely what you are doing ?

**Comment by cyberic99** [comment]
sorry it wasn't clear.
I was talkong about my application, that talks to o-s-c. At each launch, it destroys and recreates some widgets

**Comment by jean-emmanuel** [comment]
Each time widgets are recreated these message are likely to appear, however the number of messages appearing for a given operation is not supposed to increase, can you provide a message or sequence of messages that would show such behaviour ?

**Comment by jean-emmanuel** [comment]
Looks like the big memory leak got fixed :)

**Comment by cyberic99** [comment]
Well done!
will test it asap.

**Comment by jean-emmanuel** [comment]
Included in v0.23, cheers !
